lib/IDE/Utils.cpp

```diff
@@ -226,6 +227,190 @@ static std::string adjustClangTriple(StringRef TripleStr) {
   return Result;
 }
 
+static FrontendInputsAndOutputs resolveSymbolicLinksInInputs(
```

Comment:
`adjustClangTriple()` above is only used in `initInvocationByClangArguments()` below. I think you should place this block before `adjustClangTriple()`.